DEHPEcosimPro/ViewModel/Rows/ParameterDifferenceViewModel.cs

```diff
@@ -0,0 +1,324 @@
+                }
+
+                for (int i = 0; i < this.listofsetOfNewValues.Count; i++)
+                {
+                    this.ListOfParameters.Add(this.PopulateParameterDifferenceRowViewModel(i, true, false));
+                }
+            }
+            else if (!isoptiondependant && statedependance == null)
+            {
+                this.PopulateListOfSets(null, null);
+
+                for (int i = 0; i < this.listofsetOfNewValues.Count; i++)
+                {
+                    this.ListOfParameters.Add(this.PopulateParameterDifferenceRowViewModel(i, false, false));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Compute values and add it to the object
+        /// </summary>
+        /// <param name="index">number of iteration</param>
+        /// <param name="isOptions">Has the Thing Options?</param>
+        /// <param name="isState">Has the Thing States?</param>
+        /// <returns><see cref="ParameterDifferenceRowViewModel"/></returns>
+        private ParameterDifferenceRowViewModel PopulateParameterDifferenceRowViewModel(int index, bool isOptions, bool isState)
+        {
+            var setOfNewValues = this.listofsetOfNewValues[index].ActualValue;
+            var setOfOldValues = this.listofsetOfOldValues[index].ActualValue;
+
+            object NewValue;
+            object OldValue;
+            object Name = this.ModelCode();
+
+            if (setOfNewValues.Count > 1)
+            {
+                NewValue = setOfNewValues.ToString();
+            }
+            else
+            {
+                NewValue = setOfNewValues.FirstOrDefault();
+            }
+
+            if (setOfOldValues.Count > 1)
+            {
+                OldValue = setOfOldValues.ToString();
+            }
+            else
+            {
+                OldValue = setOfOldValues.FirstOrDefault();
+            }
+
+
+            if (isOptions && isState)
+            {
+                Name = Name + $"\\{this.listofsetOfNewValues[index].ActualOption.ShortName}\\{this.listofsetOfNewValues[index].ActualState.ShortName}";
+            }
+            else if (isOptions && !isState)
+            {
+                Name = Name + $"\\{this.listofsetOfNewValues[index].ActualOption.ShortName}";
+            }
+            else if (!isOptions && isState)
+            {
+                Name = Name + $"{this.listofsetOfNewValues[index].ActualState.ShortName}";
+            }
+            else if (!isOptions && !isState)
+            {
+                Name = this.ModelCode();
+            }
+
+            this.CalculateDiff(OldValue, NewValue, out string Difference, out string PercentDiff);
+
+            return new ParameterDifferenceRowViewModel(this.OldThing, this.NewThing, Name, OldValue, NewValue, Difference, PercentDiff);
+
+        }
+
+        /// <summary>
+        /// Add The valueSet to list
+        /// </summary>
+        /// <param name="option"><see cref="Option"/></param>
```

Comment:
Like in many param comments I see in this PR the class of the param is not enough for a comment.

Reply:
For instance what is the Option of? What is the State of?